opal/mca/reachable/netlink/configure.m4

```diff
@@ -74,15 +76,17 @@ AC_DEFUN([OPAL_REACHABLE_NETLINK_CHECK_LIBNL3],[
        save_CPPFLAGS=$CPPFLAGS
        save_LIBS=$LIBS
 
-       $1="-I/usr/include/libnl3"
-       CPPFLAGS="$$1 $CPPFLAGS"
-       AC_MSG_CHECKING([for /usr/include/libnl3])
-       AS_IF([test -d "/usr/include/libnl3"],
-             [AC_MSG_RESULT([present])
-              AC_CHECK_HEADER(
-                [netlink/version.h],
-                [AC_COMPILE_IFELSE(
-                    [AC_LANG_PROGRAM([[
+       AS_IF([test $opal_libnl_version -eq 1],
+             [HAVE_LIBNL3=0],
+             [$1="-I/usr/include/libnl3"
```

Comment:
Is this macro enough?

I'm wondering if we need a more complete check for libnl-vs-libnl3.  E.g., something that calls OPAL_CHECK_PACKAGE (which does very complete checks for dependent libraries).

For example, the libfabric usnic provider uses the OPAL_CHECK_PACKAGE-inspired FI_CHECK_PACKAGE to do a very complete set of libnl-vs-libnl3 checks: see https://github.com/ofiwg/libfabric/blob/master/prov/usnic/configure.m4#L143-L312

It also does some gratuitous checks, like checking to ensure that the netlink.h we found matches the library we found (not because we found a real world case where this happened, but rather because we know that libnl is evil and this is not beyond their scope of evilness...).